File: jobs/2026-05-30__07-57-35/multi_service_api_go__AYsLB6z/artifacts/user/myproject/order/order.go
package order

import (
	"context"
	"encore.app/user"
)

type Order struct {
	ID       string `json:"id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

//encore:api public path=/order/:id
func GetOrder(ctx context.Context, id string) (*Order, error) {
	// Call the user service
	u, err := user.GetUser(ctx, "user-123")
	if err != nil {
		return nil, err
	}

	return &Order{
		ID:       id,
		UserID:   u.ID,
		UserName: u.Name,
	}, nil
}

File: jobs/2026-07-03__05-39-37/encore_multi_service_api_ts__efbn4XR/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

export interface Order {
  orderId: number;
  userId: number;
  userName: string;
}

// getOrder returns an order by ID.
// It calls the user service internally to fetch the userName
// (always fetching the user with ID 1 for simplicity).
export const getOrder = api(
  { method: "GET", path: "/order/:id", expose: true },
  async (params: { id: number }): Promise<Order> => {
    // Call the user service internally to fetch the user with ID 1.
    const u = await user.getUser({ id: 1 });
    return {
      orderId: params.id,
      userId: u.id,
      userName: u.name,
    };
  }
);